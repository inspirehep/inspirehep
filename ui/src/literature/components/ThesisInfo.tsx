import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Map } from 'immutable';
import InlineList from '../../common/components/InlineList';
import Institution from './Institution';

class ThesisInfo extends Component {
  renderInstutions() {
    const { thesisInfo } = this.props;
    return (
      <InlineList
        wrapperClassName="di"
        items={thesisInfo.get('institutions')}
        extractKey={institution => institution.get('name')}
        renderItem={institution => (
          <Institution institution={institution} />
        )}
      />
    );
  }

  renderDateOrDefenseDate() {
    const { thesisInfo } = this.props;
    const defenseDate = thesisInfo.get('defense_date');
    if (defenseDate) {
      return `(defense: ${defenseDate})`;
    }

    const date = thesisInfo.get('date');
    if (date) {
      return `(${date})`;
    }

    return null;
  }

  render() {
    const { thesisInfo } = this.props;

    if (!thesisInfo) {
      return null;
    }

    const degreeType = thesisInfo.get('degree_type');
    return (
      <span>
        <span>Thesis: </span>
        {degreeType && <span>{degreeType} </span>}
        <span>{this.renderInstutions()} </span>
        <span>{this.renderDateOrDefenseDate()}</span>
      </span>
    );
  }
}

ThesisInfo.propTypes = {
  thesisInfo: PropTypes.instanceOf(Map),
};

ThesisInfo.defaultProps = {
  thesisInfo: null,
};

export default ThesisInfo;
