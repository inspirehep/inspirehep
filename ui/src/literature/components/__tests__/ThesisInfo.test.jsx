import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import ThesisInfo from '../ThesisInfo';

describe('ThesisInfo', () => {
  it('renders full thesis info', () => {
    const thesisInfo = fromJS({
      degreeType: 'phd',
      date: '11-11-2011',
      defense_date: '12-12-2012',
      institutions: [
        {
          name: 'Institution 1',
        },
      ],
    });
    const wrapper = shallow(<ThesisInfo thesisInfo={thesisInfo} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with defense date', () => {
    const thesisInfo = fromJS({
      date: '11-11-2011',
      defense_date: '12-12-2012',
    });
    const wrapper = shallow(<ThesisInfo thesisInfo={thesisInfo} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with date and without defense date', () => {
    const thesisInfo = fromJS({
      date: '11-11-2011',
    });
    const wrapper = shallow(<ThesisInfo thesisInfo={thesisInfo} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders without any date', () => {
    const thesisInfo = fromJS({
      institutions: [
        {
          name: 'Institution 1',
        },
      ],
    });
    const wrapper = shallow(<ThesisInfo thesisInfo={thesisInfo} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders empty if null', () => {
    const wrapper = shallow(<ThesisInfo />);
    expect(wrapper).toMatchSnapshot();
  });
});
