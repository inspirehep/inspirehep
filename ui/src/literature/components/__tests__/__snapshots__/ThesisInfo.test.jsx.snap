// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThesisInfo renders empty if null 1`] = `<DocumentFragment />`;

exports[`ThesisInfo renders full thesis info 1`] = `
<DocumentFragment>
  <span>
    <span>
      Thesis: 
    </span>
    <span>
      <div
        class="__InlineList__ di"
        data-testid="inline-data-list"
      >
        <ul>
          <li>
            <span>
              Institution 1
            </span>
          </li>
        </ul>
      </div>
       
    </span>
    <span>
      (defense: 12-12-2012)
    </span>
  </span>
</DocumentFragment>
`;

exports[`ThesisInfo renders with date and without defense date 1`] = `
<DocumentFragment>
  <span>
    <span>
      Thesis: 
    </span>
    <span>
       
    </span>
    <span>
      (11-11-2011)
    </span>
  </span>
</DocumentFragment>
`;

exports[`ThesisInfo renders with defense date 1`] = `
<DocumentFragment>
  <span>
    <span>
      Thesis: 
    </span>
    <span>
       
    </span>
    <span>
      (defense: 12-12-2012)
    </span>
  </span>
</DocumentFragment>
`;

exports[`ThesisInfo renders without any date 1`] = `
<DocumentFragment>
  <span>
    <span>
      Thesis: 
    </span>
    <span>
      <div
        class="__InlineList__ di"
        data-testid="inline-data-list"
      >
        <ul>
          <li>
            <span>
              Institution 1
            </span>
          </li>
        </ul>
      </div>
       
    </span>
    <span />
  </span>
</DocumentFragment>
`;
